File: src/main/java/hu/frontrider/arcana/research/ResearchEventManager.java
package hu.frontrider.arcana.research;

import hu.frontrider.arcana.research.researchevents.StartingFires;
import net.minecraftforge.common.MinecraftForge;

/**
 * @author Kis András Gábor
 * 2018.08.28.
 */
public class ResearchEventManager {
    
    public static void initHandlers()
    {
        registerhandler(new StartingFires());
    }
    
   static void registerhandler(Object handler){
        MinecraftForge.EVENT_BUS.register(handler);
    }
}

File: src/main/java/hu/frontrider/arcana/research/ResearchEventManager.kt
package hu.frontrider.arcana.research

import hu.frontrider.arcana.research.researchevents.PlantExperiments
import hu.frontrider.arcana.research.researchevents.StartingFires
import hu.frontrider.arcana.research.researchevents.TheScar
import net.minecraftforge.common.MinecraftForge

/**
 * @author Kis András Gábor
 * 2018.08.28.
 */
class ResearchEventManager {

    fun initHandlers() {
        registerhandler(StartingFires())
        registerhandler(PlantExperiments())
        registerhandler(TheScar())
    }

    internal fun registerhandler(handler: Any) {
        MinecraftForge.EVENT_BUS.register(handler)
    }
}
